framework/src/loops/ComputeMaterialsObjectThread.C

```diff
@@ -171,7 +171,15 @@ ComputeMaterialsObjectThread::onInternalSide(const Elem * elem, unsigned int sid
 {
   if (_need_internal_side_material)
   {
-    _assembly[_tid]->reinit(elem, side);
+    const Elem * neighbor = elem->neighbor_ptr(side);
+    const dof_id_type elem_id = elem->id(), neighbor_id = neighbor->id();
```

Comment:
Do we need to check if neighbor is empty?  Or this has been guarded by `_need_internal_side_material` ?

Reply:
In `ThreadedElementLoopBase` we do not call `onInternalSide` if the neighbor is null